pkg/kv/kvprober/kvprober_integration_test.go

```diff
@@ -92,6 +94,55 @@ func TestProberDoesReadsAndWrites(t *testing.T) {
 		require.Zero(t, p.Metrics().ProbePlanFailures.Count())
 	})
 
+	t.Run("happy path with tracing enabled", func(t *testing.T) {
+		// Define a temporary logging directory for the kv prober logs to be written to
+		logScope := log.Scope(t)
+		defer logScope.Close(t)
+
+		s, _, p, cleanup := initTestServer(t, base.TestingKnobs{})
+		defer cleanup()
+
+		kvprober.ReadEnabled.Override(ctx, &s.ClusterSettings().SV, true)
+		kvprober.ReadInterval.Override(ctx, &s.ClusterSettings().SV, 5*time.Millisecond)
+
+		kvprober.WriteEnabled.Override(ctx, &s.ClusterSettings().SV, true)
+		kvprober.WriteInterval.Override(ctx, &s.ClusterSettings().SV, 5*time.Millisecond)
+
+		kvprober.QuarantineEnabled.Override(ctx, &s.ClusterSettings().SV, true)
+		kvprober.QuarantineInterval.Override(ctx, &s.ClusterSettings().SV, 5*time.Millisecond)
+
+		kvprober.TracingEnabled.Override(ctx, &s.ClusterSettings().SV, true)
+
+		testutils.SucceedsSoon(t, func() error {
+			if p.Metrics().ReadProbeAttempts.Count() < int64(50) {
+				return errors.Newf("read count too low: %v", p.Metrics().ReadProbeAttempts.Count())
+			}
+			if p.Metrics().WriteProbeAttempts.Count() < int64(50) {
+				return errors.Newf("write count too low: %v", p.Metrics().WriteProbeAttempts.Count())
+			}
+			return nil
+		})
+
+		// Read the health chanel log file where the kv prober logs are stored
+		buf := new(bytes.Buffer)
```

Comment:
I'm not sure if this read-logs-from-disk approach will be approved by KV. I think we may need an assertion made against an in-memory log buffer. We may not want to depend on reads & writes done from actual disks from within tests. I'm not confident about this tho. We will see what KV says. I will do some research in the meantime about the in-memory-log-buffer approach.

Reply:
That's true. I will also have a look in the meantime. Thank you!

Reply:
I talked to David from o11y team, and he says the pattern to use in the CRDB repo is to implement an "interceptor". API here: https://github.com/cockroachdb/cockroach/blob/045acd250684d5ca2497478e34a84c4ad1d5699b/pkg/util/log/intercept.go#L26. Example usage here: https://github.com/cockroachdb/cockroach/blob/eccb77d2320955b477bb03bc1f1aee137bd1de76/pkg/ccl/telemetryccl/telemetry_logging_test.go#L158

Reply:
Done now!